### ISSUE TITLE
ParseArgs() should no longer return err if user asked for help

### ISSUE THREAD

**Issue by thrawn01** [thrawn01/args#28]
Implementation should simply check for help option
```go
opts, err := parser.ParseArgs(nil)
// always check for help first, user might omit 
// a required argument but still wants help
if opts.Bool("help") {
    parser.PrintHelp()
    os.Exit(0)
}
if err != nil {
   fmt.Fprintf(os.Stderr, "-- %s\n", err)
   os.Exit(1)
}

```

**Comment by thrawn01** [comment]
`ParseArgs()` should never return nil

**Comment by thrawn01** [comment]
Implemented in #33
